Implement matching service

GenFusion122 opened the pull request:
Implementation of the backend for the matching-service that allows users to queue and be matched based on `difficulty` and `language`.

`enqueue` - enqueues the user making the request.
`dequeue` - removes the user from the queue.
`getLength` - retrieves the length of the queue, mainly for supporting purposes.

GenFusion122 committed:
Implement matching service

Implementation of the matching-service that allows users to queue and be matched based on `difficulty` and `language`.

`enqueue` - enqueues the user making the request
`dequeue` - removes the user from the queue
`getLength` - retrieves the length of the queue, mainly for supporting purposes

chia-yh reviewed (approved):
LGTM, but might be some disjoint between behavior and name of `enqueue` and `dequeue`, thought its just a nit

cjyothika commented on backend/matching-service/controllers/matching.controller.js:
```diff
@@ -0,0 +1,61 @@
+
+exports.dequeue = (req, res) => {
+    try {
+        let existingMatch = mQueue.checkUser(match);
```
Can you point me to where match is defined in this method?

cjyothika replied:
Also, maybe the type of match should be checked?

GenFusion122 replied:
Yeah it's not defined here my b, lemme fix it.

cjyothika reviewed (commented):
Generally, it looks good but I have a few questions.

GenFusion122 committed:
Minor issue fix

match was undefined in the dequeue function, resolved it by writing a new function that took in a userid instead of a match object.

GenFusion122 committed:
Minor tweaks

Changed when the code creates a new match object and replaced the checkUser function entirely with checkUserId

chia-yh reviewed (approved):
LGTM

cjyothika reviewed (approved):
LGTM
